sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -872,9 +873,8 @@ ssize_t coap_opt_add_uquery2(coap_pkt_t *pkt, const char *key, size_t key_len,
         qs[key_len] = '=';
         memcpy(&qs[key_len + 1], val, val_len);
     }
-    qs[qs_len] = '\0';
```

Comment:
Why don't we need this anymore? I took a quick look at `_add_opt_pkt` and `_put_delta_optlen` but cannot find where terminating the string with a `\0` is done.

Reply:
We don't write the null terminator to the option value because `_add_opt_pkt()` requires the length of the option value, so it knows exactly how many bytes to copy from `qs`.